src/api/order/services/post.js

```diff
@@ -0,0 +1,30 @@
+import Order from '../../../models/order';
+import { sendSuccessResponse, sendErrorResponse } from '../../../utils/responseHelpers';
+
+async function createOrder(req, res) {
+  const { productList } = req.body;
+  const { orderProducts, user } = req;
+  const order = { userId: user._id, productList: [] };
+  const updatedProducts = [];
+  let allPrice = 0;
+  let quantityOrderItem;
+
+  orderProducts.forEach((elem, i) => {
+    order.productList[i] = { id: elem._id, price: elem.price };
+    quantityOrderItem = productList.find((item) => item.id === order.productList[i].id.toString());
```

Comment:
Do we need to find by id? productList and orderProducts arrays are identical, so you can get by `productList[i]` instead of finding. And why don't you use map, instead of for each?
```
order.productList = orderProducts.map(item, i) => {
...
```

Even if you don't use map, you must not add a new object in array with `order.productList[i]`. It's better just to `push`